File: datasets/lua/HumanEval_7_filter_by_substring.lua
-- Filter an input list of strings only for ones that contain given substring
-- >>> filter_by_substring([], 'a')
-- []
-- >>> filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'a')
-- ['abc', 'bacd', 'array']
local function filter_by_substring(strings, substring)
  local filtered = {}
  for _, string in ipairs(strings) do
    if string:find(substring) then
      table.insert(filtered, string)
    end
  end
  return filtered
end
-- Unit tests below

lu = require('luaunit')

function test_humaneval()
local candidate = filter_by_substring
    lu.assertEquals(candidate({}, 'john'), {})
    lu.assertEquals(candidate({'xxx', 'asd', 'xxy', 'john doe', 'xxxAAA', 'xxx'}, 'xxx'), {'xxx', 'xxxAAA', 'xxx'})
    lu.assertEquals(candidate({'xxx', 'asd', 'aaaxxy', 'john doe', 'xxxAAA', 'xxx'}, 'xx'), {'xxx', 'aaaxxy', 'xxxAAA', 'xxx'})
    lu.assertEquals(candidate({'grunt', 'trumpet', 'prune', 'gruesome'}, 'run'), {'grunt', 'prune'})
end

os.exit(lu.LuaUnit.run())

File: datasets/rs/HumanEval_7_filter_by_substring.rs
/// Filter an input list of strings only for ones that contain given substring
/// >>> filter_by_substring([], 'a')
/// []
/// >>> filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'a')
/// ['abc', 'bacd', 'array']
fn filter_by_substring(strings: Vec<String>, substring: String) -> Vec<String> {
    strings
        .into_iter()
        .filter(|s| s.contains(&substring))
        .collect()

}

fn main() {
    let candidate = filter_by_substring;
    assert_eq!(candidate(Vec::<String>::new(), String::from("john")), Vec::<String>::new());
    assert_eq!(candidate(vec![String::from("xxx"), String::from("asd"), String::from("xxy"), String::from("john doe"), String::from("xxxAAA"), String::from("xxx")], String::from("xxx")), vec![String::from("xxx"), String::from("xxxAAA"), String::from("xxx")]);
    assert_eq!(candidate(vec![String::from("xxx"), String::from("asd"), String::from("aaaxxy"), String::from("john doe"), String::from("xxxAAA"), String::from("xxx")], String::from("xx")), vec![String::from("xxx"), String::from("aaaxxy"), String::from("xxxAAA"), String::from("xxx")]);
    assert_eq!(candidate(vec![String::from("grunt"), String::from("trumpet"), String::from("prune"), String::from("gruesome")], String::from("run")), vec![String::from("grunt"), String::from("prune")]);
}
